fix(core): Prevent shared user details being saved alongside execution data

freyamade opened the pull request.

Commit:
:hammer: - Remove `shared` key from execution save data

Commit:
:shirt: - Using import type where needed

netroy commented on packages/cli/src/WorkflowExecuteAdditionalData.ts:
```diff
@@ -583,13 +584,19 @@ function hookFunctionsSave(parentProcessMode?: string): IWorkflowExecuteHooks {
+					// As a result, we should remove any sensitive information before saving to the database
+					const { shared, ...workflowData } = this.workflowData as WorkflowEntity;
+
+					console.log(this.workflowData, '\n', workflowData);
```
please remove the console.log

freyamade replied:
good catch ty

Commit:
remove console.log

Commit:
:hammer: - Create new clean workflowData instead of removing shared

If IWorkflowBase changes in future, TS will error out here ensuring it's kept up to date

netroy commented on packages/cli/src/WorkflowExecuteAdditionalData.ts:
```diff
@@ -583,13 +584,28 @@ function hookFunctionsSave(parentProcessMode?: string): IWorkflowExecuteHooks {
 
+					// Although it is treated as IWorkflowBase here, it's being instantiated elsewhere with properties that may be sensitive
+					// As a result, we should create an IWorkflowBase object with only the data we want to save in it.
+					const pristineWorkflowData: IWorkflowBase = {
```
you could use `lodash.pick` or `lodash.omit` to make this less verbose.

Commit:
:hammer: - use lodash.pick for less verbosity

Commit:
:hammer: - fix lodash imports

krynble reviewed (approved):
LGTM

janober commented:
Got released with `n8n@0.214.0`
